service/prometheus/update_test.go

```diff
@@ -0,0 +1,404 @@
+package prometheus
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+)
+
+// Test_Prometheus_isManaged tests the isManaged function.
+func Test_Prometheus_isManaged(t *testing.T) {
+	tests := []struct {
+		scrapeConfig config.ScrapeConfig
+		isManaged    bool
+	}{
+		{
+			scrapeConfig: config.ScrapeConfig{},
+			isManaged:    false,
+		},
+
+		{
+			scrapeConfig: config.ScrapeConfig{
+				JobName: "xa5ly",
+				ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
+					StaticConfigs: []*config.TargetGroup{
+						{
+							Targets: []model.LabelSet{
+								model.LabelSet{"apiserver.xa5ly": ""},
+							},
+						},
+					},
+				},
+			},
+			isManaged: false,
+		},
+
+		{
+			scrapeConfig: config.ScrapeConfig{
+				JobName: "xa5ly",
+				ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
+					StaticConfigs: []*config.TargetGroup{
+						{
+							Targets: []model.LabelSet{
+								model.LabelSet{"apiserver.xa5ly": ""},
+							},
+							Labels: model.LabelSet{ClusterLabel: ""},
+						},
+					},
+				},
+			},
+			isManaged: true,
+		},
+	}
+
+	for index, test := range tests {
+		returnedIsManaged := isManaged(test.scrapeConfig)
+
+		if test.isManaged != returnedIsManaged {
+			t.Fatalf(
+				"%d: incorrect managed: expected: %t, received: %t, for: \n%s",
+				index,
+				test.isManaged,
+				returnedIsManaged,
+				spew.Sdump(test.scrapeConfig),
+			)
+		}
+	}
+}
+
+// Test_Prometheus_UpdateConfig tests the UpdateConfig function.
+func Test_Prometheus_UpdateConfig(t *testing.T) {
+	tests := []struct {
+		config        config.Config
+		scrapeConfigs []config.ScrapeConfig
+
+		expectedConfig config.Config
+	}{
+		// Test an empty config, and one scrape config,
```

Comment:
I would like to see test numbers.

Reply:
The test numbers really bug me, actually. I find I have to re-jig them all the time. Blocker?

Reply:
No blocker but I find them very useful as soon as test cases go beyond a single visible page.